src/ducks/SANCharts/ChartPage.js

```diff
@@ -229,6 +241,12 @@ class ChartPage extends Component {
     this.setState(prev => ({ isAdvancedView: !prev.isAdvancedView }))
   }
 
+  componentDidUpdate (prevProps, prevState, snapshot) {
+    if (!isEqual(prevProps, this.props)) {
+      this.setState(getChartInitialState(this.props))
+    }
+  }
```

Comment:
Why, just why?
Component should not be implemented as the **controlled (via props) component** and **uncontrolled (stateful) component** at the same time. Not mentioning that additionally to the React reconciliation this code adds deep equality check

Reply:
@DmitriVanGuard , but if this component used with custom properties? It has inner suggestion component for new selected project, but it also could be changed outside by selected props. And then we need to change a state with new project, selected outside and passed via props

Reply:
What are those `custom properties`?
You described the problem of the architecture. It's should be a part of the chart widget.
![image](https://user-images.githubusercontent.com/25135650/63492463-3a295e80-c4c2-11e9-9eb3-ed4e1de5a26f.png)

Reply:
Ok, will move as optional header of widget

Reply:
@DmitriVanGuard , please, check. I add custom 'headerComponent' which could be passed into chartwidget.js